Sources/_SwiftSyntaxTestSupport/IncrementalParseTestUtils.swift

```diff
@@ -0,0 +1,127 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftParser
+import XCTest
+
+/// This function is used to verify the correctness of incremental parsing
+/// containing three parts:
+/// 1. Round-trip on the incrementally parsed syntax tree.
+/// 2. verify that incrementally parsing the edited source base on the original source produces the same syntax tree as reparsing the post-edit file from scratch.
+/// 3. verify the reused nodes fall into expectations.
+public func verifyIncrementalParse(
+  originalSource: String,
+  editedSource: String,
+  concurrentEdits: ConcurrentEdits,
```

Comment:
What do you think about passing the marked source to the `verifyIncrementalParse` function and running `extractMarkersForIncrementParseTest` from here. That should simplify the assert invocations.

Reply:
That's because in the previous design, we have to zip the re-used range and re-used nodes in the test case on our own. So I just separate this two functions.

I have changed this in the new commit.